src/core/name.cpp

```diff
@@ -23,19 +23,21 @@ Name::Name(const std::string& first, const std::string& last)
 {
     chocan_user_exception::Info errors;
 
-    ( !Validators::length(first, 1, 25) ) 
-        ? errors.push_back("First name cannot be empty")
-        : void();
-    ( !Validators::length(last, 1, 25) ) 
-        ? errors.push_back("Last name cannot be empty")
-        : void();
-    ( !Validators::length(first + last, 1, 25) ) 
-        ? errors.push_back("Full name must be less than 25 characters")
-        : void();
-
-    ( !errors.empty() ) 
-        ? throw invalid_name("Invalid name length", errors)
-        : void();
+    if(!Validators::length(first + last, 1, 25) || !Validators::length(first,1,24) || !Validators::length(last,1,24)) {
```

Comment:
Same comments about the new exception design as I had about the address class. Also you're performing the same validators twice in these conditionals.

Reply:
true, but in my understanding all needed answers for the conditional are evaluated at once. the relation between the first,last,full name length is know after the first conditional, all of which we need to determine regardless. then it processes the three cases at once without recomputing the expressions, pipelined but you get what i mean. it seems though and in either case ill fix this because it wont be using this logic after the refactor.

Reply:
this no longer accurately represents the state of the branch